### ISSUE TITLE
Rule Change: `no-warning-comments`: Add `ignorePattern`

### ISSUE THREAD

**Issue by wedi** [eslint/eslint#15133]
### What rule do you want to change?

no-warning-comments

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
ignorePattern: [/https:\/\/app.asana.com\/[0-9]+\/[0-9]+/, / [A-Z]{2,}-[0-9]+/, /🧞‍♂️/]
// TODO: Todo without ticket reference will break
// TODO: Todo with Asana ticket url will not break https://app.asana.com/0/4711/list
// TODO: Neither will this Jira one ESLINT-123
// FIXME: Neither will this when your config allows the magical 🧞‍♂️ emoji.
```

### What does the rule currently do for this code?

It will complain about all rules because the comments contain the terms `TODO` and/or `FIXME` (using default settings).

### What will the rule do after it's changed?

* It will not complain if a new regex Option matches the comment.
* It will allow to check for my requirement to always add ticket references to such comments.
* It will allow to standardize the wording of these kinds of comments.
* It will allow anything that matches your regex.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Would you accept such a PR or should I write my own?

Is there a convention for such "ignore rules"? What should the name of such option be? `max-len` uses [`ignorePattern`](https://eslint.org/docs/rules/max-len#ignorepattern) but it does not allow for arrays.

`max-len` also has an additional option [`ignoreUrls`](https://eslint.org/docs/rules/max-len#ignoreurls) that would match my personal requirements here, too. Should I add both?

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by wedi** [comment]
Thanks for looking into this @nzakas.
I actually read that readme and I am surprised that you consider `no-warning-comments` to be a _stylistic_ rule. Please expand on the reasoning behind that. I would like to understand it.
